library/common/engine.cc

```diff
@@ -106,11 +107,14 @@ Engine::~Engine() {
   main_thread_.join();
 }
 
-envoy_status_t Engine::recordCounterInc(const std::string& elements, uint64_t count) {
-  if (server_ && client_scope_) {
+envoy_status_t Engine::recordCounterInc(
```

Comment:
Currently in this pr the change is only applied to counter, after a green light of this pr by everyone, I'll expand the implementation to gauge and histogram.